### DISCUSSION TITLE
Regarding Smart Contract of Lesson 7 Challenge !

### DISCUSSION THREAD

**Discussion by heysourin** [Cyfrin/foundry-full-course-cu#96] [Q&A] [answered]
I have solved this level. But I have a question regarding the smart contract.

I don't see any state variable in the smart contract, to be stored in the storage slot, how did the question-setter added a data to slot 777?
Can anyone please clarify? Thank you!!

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/98945276/902010eb-fdc5-42ef-a914-a2a6a19c45df)

**Comment by ghost** [comment, marked as answer]
Hello @heysourin I really don't know this challenge but I can tell you that `assembly` gives you low-level control over storage e.g. writing to the slot you want.

You don't need a previously declared state variable at slot `777` to write there, with `sstore` you can write there directly and fill the slot with the number.

You only need to pass the params to `sstore` and then I'm guessing `AFoundryChallengeCourse` has a function that reads from that slot and check if the value is `777`.

**Reply by heysourin** [reply]
Thank you!! @pacelliv

**Comment by cromewar** [comment]
Can you help me with this one @alymurtazamemon

**Reply by alymurtazamemon** [reply]
@cromewar Yes, do you know how to read the data from any storage slot?

**Reply by PatrickAlphaC** [reply]
[hint](https://book.getfoundry.sh/reference/cast/cast-storage?highlight=storage#cast-storage)

**Comment by 0xTagger** [comment]
hey @heysourin any hints you can provide on how to solve the challenge? been stuck on it for a while. Thanks!

**Comment by SCPassion** [comment]
I am also stuck with this challenge. Initially, the contract assign value 123 to location 777. To my understanding, if I put 123 to input "valueAtStorageLocationSevenSevenSeven", it should pass. However, it is not. Can anyone give some hints?

**Reply by 0xTagger** [reply]
Hey @SCPassion - this post and hint helped me figure out the answer! give it a look and let me know if you need more assistance. https://github.com/Cyfrin/foundry-full-course-f23/discussions/277#discussioncomment-6379333

**Reply by SCPassion** [reply]
Thanks.
